packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -1,35 +1,348 @@
+    seed
+  );
+
+  const { certified, declared, verified } = await getAttributeInvolved(
+    consumer,
+    seed,
+    attributeQuery
+  );
+
+  return {
+    todayDate: dateAtRomeZone(today),
+    todayTime: timeAtRomeZone(today),
+    agreementId: agreement.id,
+    submitter,
+    submissionDate: dateAtRomeZone(submissionTimestamp),
+    submissionTime: timeAtRomeZone(submissionTimestamp),
+    activator,
+    activationDate: dateAtRomeZone(activationTimestamp),
+    activationTime: timeAtRomeZone(activationTimestamp),
+    eServiceName: eservice.name,
+    producerText,
+    consumerText,
+    certifiedAttributes: getCertifiedAttributeHtml(certified),
+    declaredAttributes: getDeclaredAttributeHtml(declared),
+    verifiedAttributes: getVerifiedAttributeHtml(verified),
+  };
+};
+
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export const contractBuilder = (
-  selfcareId: SelfcareId,
   attributeQuery: AttributeQuery,
-  storeFile: FileManager["storeBytes"],
+  pdfGenerator: PDFGenerator,
+  fileManager: FileManager,
+  config: AgreementProcessConfig,
   logger: Logger
-) => ({
-  createContract: async (
-    agreement: Agreement,
-    eservice: EService,
-    consumer: Tenant,
-    producer: Tenant,
-    seed: UpdateAgreementSeed
-  ): Promise<ApiAgreementDocumentSeed> =>
-    await pdfGenerator.createDocumentSeed(
-      selfcareId,
-      agreement,
-      eservice,
-      consumer,
-      producer,
-      seed,
-      attributeQuery,
-      storeFile,
-      logger
-    ),
-});
+) => {
+  const filename = fileURLToPath(import.meta.url);
+  const dirname = path.dirname(filename);
+
+  return {
+    createContract: async (
+      selfcareId: SelfcareId,
+      agreement: Agreement,
+      eservice: EService,
+      consumer: Tenant,
+      producer: Tenant,
+      seed: UpdateAgreementSeed
+    ): Promise<ApiAgreementDocumentSeed> => {
+      const templateFilePath = path.resolve(
+        dirname,
+        "..",
+        "resources/templates/documents",
+        "agreementContractTemplate.html"
+      );
+
+      const pdfPayload = await getPdfPayload(
+        selfcareId,
+        agreement,
+        eservice,
+        consumer,
+        producer,
+        seed,
+        attributeQuery
+      );
+
+      const documentId = generateId<AgreementDocumentId>();
+      const documentName = createAgreementDocumentName(
+        agreement.consumerId,
+        agreement.producerId
+      );
+
+      const pdfBuffer: Buffer = await pdfGenerator.generate(
+        templateFilePath,
+        pdfPayload
+      );
+
+      const documentPath = await fileManager.storeBytes(
+        config.s3Bucket,
+        `${config.agreementContractsPath}/${agreement.id}`,
+        documentId,
+        documentName,
+        pdfBuffer,
+        logger
+      );
+
+      return {
+        id: documentId,
+        name: documentName,
+        contentType: "application/pdf",
+        prettyName: "Richiesta di fruizione",
```

Comment:
Could we have these two defined as constants at the beginning of the file? Just to not hardcode them here inside the PDF creation logic

Reply:
Added with commit (20df5c69b83edb88dec90e55d372b32654219abc)[https://github.com/pagopa/interop-be-monorepo/pull/520/commits/20df5c69b83edb88dec90e55d372b32654219abc]